odbc/src/api/functions.rs

```diff
@@ -1365,20 +1365,26 @@ pub unsafe extern "C" fn SQLFetch(statement_handle: HStmt) -> SqlReturn {
             match res {
                 Err(e) => {
                     stmt.errors.write().unwrap().push(e.into());
-                    return SqlReturn::ERROR;
+                    SqlReturn::ERROR
                 }
-                Ok(b) => {
+                Ok((b, e)) => {
                     let mut stmt_attrs = stmt.attributes.write().unwrap();
+                    if let Some(error) = e.clone() {
+                        stmt.errors.write().unwrap().push(ODBCError::Core(error));
```

Comment:
I don't think that this will work.
This will report a General Error in most cases, whereas we want to return a General Warning.
You need to add the SQL state GENERAL_WARNING (01000) and report this instead.